src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -163,6 +168,76 @@ protected ExecutionStrategy(DataFetcherExceptionHandler dataFetcherExceptionHand
      */
     public abstract CompletableFuture<ExecutionResult> execute(ExecutionContext executionContext, ExecutionStrategyParameters parameters) throws NonNullableFieldWasNullException;
 
+    public CompletableFuture<Object> executeObject(ExecutionContext executionContext, ExecutionStrategyParameters parameters) throws NonNullableFieldWasNullException {
+        Instrumentation instrumentation = executionContext.getInstrumentation();
+        InstrumentationExecutionStrategyParameters instrumentationParameters = new InstrumentationExecutionStrategyParameters(executionContext, parameters);
+        ExecutionStrategyInstrumentationContext executionStrategyCtx = ExecutionStrategyInstrumentationContext.nonNullCtx(instrumentation.beginExecutionStrategy(instrumentationParameters, executionContext.getInstrumentationState()));
+
+        MergedSelectionSet fields = parameters.getFields();
+        List<String> fieldNames = fields.getKeys();
+        Async.CombinedBuilder<FieldValueInfo> futures = resolveFieldsWithInfo(executionContext, parameters, fields, fieldNames);
+
+        CompletableFuture<Object> overallResult = new CompletableFuture<>();
+        CompletableFuture<ExecutionResult> instrumentationFuture = overallResult.thenApply(val -> ExecutionResult.newExecutionResult().data(val).build());
+        executionStrategyCtx.onDispatched(instrumentationFuture);
```

Comment:
This line here is a  backwards compat hack and should be removed

```
CompletableFuture<ExecutionResult> instrumentationFuture = overallResult.thenApply(val -> ExecutionResult.newExecutionResult().data(val).build());
```

Its allocated a ER so that I didnt have to change the `Instrumentation` signature but really I think if we want to do this code we should change the signatures.  So `Instrumentation` is the biggest breaking change here in effect